### PR TITLE
Don't show archived products in assign production to location UI

### PR DESCRIPTION
**Pull request by sravfeyn**
http://manage.dimagi.com/default.asp?189590#1062111
@calellowitz @kaapstorm 

I did not test this (I couldn't figure out right combination of toggles and previews to enable), but this is a very simple change.

### CODE REVIEW

File: corehq/apps/locations/views.py [hunk 1]
```diff
@@ -474,9 +474,9 @@ def users_form(self):
         return form
 
     @property
-    def all_products(self):
+    def active_products(self):
```
Comment (sravfeyn) [inline corehq/apps/locations/views.py]: I have made sure this is not being used anywhere else

File: corehq/apps/locations/views.py [hunk 2]
```diff
@@ -474,9 +474,9 @@ def users_form(self):
...
         return [(p.product_id, p.name)
-                for p in SQLProduct.by_domain(self.domain)]
+                for p in SQLProduct.objects.filter(domain=self.domain, is_archived=False).all()]
```
Comment (czue) [inline corehq/apps/locations/views.py]: out of scope, but i wonder if we should add a new manager or classmethod for this. seems like it should almost be the default. cc @esoergel

**Comment by czue** [comment]
:+1: when tests run.

File: corehq/apps/locations/views.py (hunk 2 above)
Comment (esoergel) [inline corehq/apps/locations/views.py]: Agreed.  That should have been made the default as part of introducing the archival workflow.  I was looking at a similar area recently and made a note to switch this over when I get the chance.

**Pull request merged**